shim/opentelemetry-opentracing-shim/src/opentelemetry/shim/opentracing_shim/__init__.py

```diff
@@ -679,9 +679,12 @@ def start_span(
             "attributes": tags,
             "start_time": start_time_ns,
         }
-        if util.opentracing_to_opentelemetry_kind_tag(tags) is not None:
-            kwargs["kind"] = util.opentracing_to_opentelemetry_kind_tag(tags)
-            del tags[SPAN_KIND]
+
+        if tags is not None and SPAN_KIND in tags:
+            if util.opentracing_to_otel_kind_tag(tags[SPAN_KIND]) is not None:
+                kwargs["kind"] = util.opentracing_to_otel_kind_tag(
+                    tags.get(SPAN_KIND, None)
+                )
```

Comment:
technically you checked 2 lines above that `SPAN_KIND` is present in `tags`, you could just:
```suggestion
                kwargs["kind"] = util.opentracing_to_otel_kind_tag(tags[SPAN_KIND])
```
(and you called it that way in the if)

similarly to remove the repetition of the call to `opentracing_to_otel_kind_tag` you could save itsoutput and use it if not None ?

Reply:
Return the default INTERNAL span kind from this `opentracing_to_otel_kind_tag` and you can get rid of the null check.